### PR TITLE
Added contract field to device type model

### PR DESCRIPTION
**Pull request by mehalter**
Change-type: minor
Signed-off-by: Micah Halter <micah@balena.io>

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

**Comment by mehalter** [comment]
I think we should be able to move the `contract` field on `Releases` to a `Contract` type as well, but I'm not sure if that would break anything at the moment or where it's used to go check

File: lib/types/contract.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export interface Contract {
```
Comment (thgreasi) [inline lib/types/contract.ts:1]: Is this of the same format like the one we accept in the releases & the hostApps?
If not, then it might better be names like `DeviceTypeContract`.
Other than that LGTM (would be nice to have more accurate types than just `object` though). Feel free to self-certify.

File: lib/types/models.ts [hunk 2]
```diff
@@ -380,6 +381,7 @@ export interface DeviceType {
 	name: string;
 	is_private: boolean;
 	logo: string | null;
+	contract: Contract | null;
```
Comment (thgreasi) [inline lib/types/models.ts:384]: I also just realized that the types of the image.contract & release.contract are `string | null`, which I guess are wrong...
Would be nice to fix them in a separate patch release.

File: lib/types/contract.ts (hunk 1 above)
Comment (mehalter) [reply lib/types/contract.ts:1]: This is the same format for all contracts based on the current contract specification. This is also why some of the types are objects

File: lib/types/models.ts (hunk 2 above)
Comment (mehalter) [reply lib/types/models.ts:384]: Yeah, I noticed this. I'll fix these in a separate patch as well

File: lib/types/contract.ts (hunk 1 above)
Comment (thgreasi) [reply lib/types/contract.ts:1]: I'm not sure why this explains the why some props are `object`s, but we definitely don't need to block this PR any more.

**Review by thgreasi — APPROVED** [review]
